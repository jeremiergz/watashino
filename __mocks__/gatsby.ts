import Gatsby from 'gatsby';

module.exports = {
  ...Gatsby,
  graphql: jest.fn(),
  navigate: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
