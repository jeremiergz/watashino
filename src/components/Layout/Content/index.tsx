import React, { HTMLAttributes } from 'react';
import Box from '../../primitives/Box';
import Heading from '../../widgets/Heading';
import SEO, { SEOProps } from '../../widgets/SEO';

type ContentProps = Pick<HTMLAttributes<HTMLElement>, 'children'> &
    Omit<SEOProps, 'title'> & {
        title?: string;
        type: 'post' | 'section';
    };

const typesMapping = {
    post: 'article',
    section: 'section',
} as const;

const Content = ({ children, description, keywords, lang, meta, title, type }: ContentProps) => {
    return (
        <>
            <SEO description={description} keywords={keywords} lang={lang} meta={meta} title={title} />
            {title && (
                <Heading marginTop={0} variant="h1">
                    {title}
                </Heading>
            )}
            <Box as={typesMapping[type]}>{children}</Box>
        </>
    );
};

Content.displayName = 'Content';

export default Content;
