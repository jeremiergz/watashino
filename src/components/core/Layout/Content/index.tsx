import React, { HTMLAttributes } from 'react';
import Heading from '../../../common/Heading';
import SEO, { SEOProps } from '../../../common/SEO';
import Box from '../../../primitives/Box';

type ContentProps = Pick<HTMLAttributes<HTMLElement>, 'children'> &
    Omit<SEOProps, 'title'> & {
        title?: string;
        type: 'post' | 'section';
    };

const typesMapping = {
    post: 'article',
    section: 'section',
} as const;

const Content = ({ children, description, keywords, lang, meta, title, type }: ContentProps) => {
    return (
        <>
            <SEO description={description} keywords={keywords} lang={lang} meta={meta} title={title} />
            {title && (
                <Heading marginTop={0} variant="h1">
                    {title}
                </Heading>
            )}
            <Box as={typesMapping[type]}>{children}</Box>
        </>
    );
};

Content.displayName = 'Content';

export default Content;
