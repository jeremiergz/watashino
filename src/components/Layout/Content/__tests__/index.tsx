import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Content from '..';

describe('components/Layout/Content Test Suite', () => {
    beforeAll(() => {
        const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
        useStaticQuery.mockImplementation(() => ({
            site: {
                siteMetadata: {
                    title: 'SEO Test title',
                    description: 'SEO Test description.',
                    author: 'SEO Test author',
                },
            },
        }));
    });

    it('matches snapshot', () => {
        const jsx = (
            <Content description="SEO Test description" keywords={['test']} lang="en" title="SEO Test">
                Content Component
            </Content>
        );
        expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
});
