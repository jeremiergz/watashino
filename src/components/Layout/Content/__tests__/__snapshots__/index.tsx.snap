// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout/Content Test Suite matches snapshot 1`] = `
Array [
  <span
    className="sc-AxjAm sc-AxirZ iFVgsB"
  >
    SEO Test
  </span>,
  <article
    className="sc-AxiKw cBoXSN"
  >
    Content Component
  </article>,
]
`;
